tests/SpecTests/SearchIndexSpecTest.php

```diff
@@ -0,0 +1,190 @@
+            ['name' => $name, 'comment' => 'Index creation test'],
+        );
+        $this->assertSame($name, $createdName);
+
+        [$index] = $this->waitForIndexes($collection, fn ($indexes) => $this->allIndexesAreQueryable($indexes));
+
+        $this->assertSame($name, $index->name);
+
+        // Convert to JSON to compare nested associative arrays and nested objects
+        $this->assertJsonStringEqualsJsonString(
+            json_encode($mapping, JSON_THROW_ON_ERROR),
+            json_encode($index->latestDefinition, JSON_THROW_ON_ERROR),
+        );
+    }
+
+    public function testCreateMultipleIndexesInBatch(): void
+    {
+        $collection = $this->createCollection($this->getDatabaseName(), $this->getCollectionName());
+        $names = ['test-search-index-1', 'test-search-index-2'];
+        $mapping = ['mappings' => ['dynamic' => false]];
+
+        $createdNames = $collection->createSearchIndexes([
+            ['name' => $names[0], 'definition' => $mapping],
+            ['name' => $names[1], 'definition' => $mapping],
+        ]);
+        $this->assertSame($names, $createdNames);
+
+        $indexes = $this->waitForIndexes($collection, fn ($indexes) => $this->allIndexesAreQueryable($indexes));
+
+        foreach ($names as $key => $name) {
+            $index = $indexes[$key];
+            $this->assertSame($name, $index->name);
+
+            // Convert to JSON to compare nested associative arrays and nested objects
+            $this->assertJsonStringEqualsJsonString(
+                json_encode($mapping, JSON_THROW_ON_ERROR),
+                json_encode($index->latestDefinition, JSON_THROW_ON_ERROR),
+            );
+        }
+    }
+
+    public function testDropSearchIndexes(): void
+    {
+        $collection = $this->createCollection($this->getDatabaseName(), $this->getCollectionName());
+        $name = 'test-search-index';
+        $mapping = ['mappings' => ['dynamic' => false]];
+
+        $createdName = $collection->createSearchIndex(
+            $mapping,
+            ['name' => $name],
+        );
+        $this->assertSame($name, $createdName);
+
+        $this->waitForIndexes($collection, fn ($indexes) => $this->allIndexesAreQueryable($indexes));
+
+        $collection->dropSearchIndex($name);
+
+        $indexes = $this->waitForIndexes($collection, fn (array $indexes): bool => count($indexes) === 0);
+
+        $this->assertCount(0, $indexes);
+    }
+
+    public function testUpdateSearchIndex(): void
+    {
+        $collection = $this->createCollection($this->getDatabaseName(), $this->getCollectionName());
+        $name = 'test-search-index';
+        $mapping = ['mappings' => ['dynamic' => false]];
+
+        $createdName = $collection->createSearchIndex(
+            $mapping,
+            ['name' => $name],
+        );
+        $this->assertSame($name, $createdName);
+
+        $this->waitForIndexes($collection, fn ($indexes) => $this->allIndexesAreQueryable($indexes));
```

Comment:
Should this also assert that an index was created? This seem related to the comment I left on the third prose test.